### DISCUSSION TITLE
The user agent does not support public key credentials.

### DISCUSSION THREAD

**Discussion by yuejunfei** [Web3Auth/Web3Auth#805] [Help]
Hello, I use Web3Auth to access the Android example, and open https://sdk.openlogin.com/login# with the browser of the Android system. It prompts me that The user agent does not support public key credentials. How can I solve this error?

**Comment by shahbaz17** [comment]
Hey @yuejunfei 

Could you share which example you are using? Or the code snippet of your android app?

And please share the error(screenshot) by going to the `chrome://inspect` tool.

**Reply by yuejunfei** [reply]
`class MainActivity : AppCompatActivity() {

    private lateinit var web3Auth: Web3Auth

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        web3Auth = Web3Auth(
            Web3AuthOptions(
                context = this,
                clientId = getString(R.string.web3auth_project_id),
                network = Web3Auth.Network.TESTNET,
                redirectUrl = Uri.parse("art.mints.myapplication://auth"),
                whiteLabel = WhiteLabelData(  // Optional param
                    "Web3Auth Sample App", null, null, "en", true,
                    hashMapOf(
                        "primary" to "#123456"
                    )
                )
            )
        )


        web3Auth.setResultUrl(intent?.data)


        // Calls sessionResponse() to check for any existing session.
        val sessionResponse: java8.util.concurrent.CompletableFuture<Web3AuthResponse> =
            web3Auth.sessionResponse()
        sessionResponse.whenComplete { loginResponse, error ->
            if (error == null) {
                Log.d("MainActivity_Web3Auth", loginResponse.toString())
            } else {
                Log.d("MainActivity_Web3Auth", error.message ?: "Something went wrong")
                // Ideally, you should initiate the login function here.
            }
        }

        // Setup UI and event handlers
        val signInButton = findViewById<TextView>(R.id.signInButton)
        signInButton.setOnClickListener { signIn() }

    }

    override fun onNewIntent(intent: Intent?) {
        super.onNewIntent(intent)

        // Handle user signing in when app is active
        web3Auth.setResultUrl(intent?.data)
    }

    private fun signIn() {
        val selectedLoginProvider = Provider.GOOGLE   // Can be GOOGLE, FACEBOOK, TWITCH etc.
        val loginCompletableFuture: java8.util.concurrent.CompletableFuture<Web3AuthResponse> =
            web3Auth.login(LoginParams(selectedLoginProvider))

        loginCompletableFuture.whenComplete { loginResponse, error ->
            if (error == null) {
                // render logged in UI
                println(loginResponse)
            } else {
                // render error UI
            }
        }
    }


}`

**Reply by grvgoel81** [reply]
Hi  @yuejunfei, please share screenshot for the prompt you are getting.
